pipeline-model-extensions/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/agent/DeclarativeAgentScript.java

```diff
@@ -27,7 +27,7 @@
 import org.jenkinsci.plugins.pipeline.modeldefinition.withscript.WithScriptScript;
 import org.jenkinsci.plugins.workflow.cps.CpsScript;
 
-
+@Deprecated
```

Comment:
Implemented in a few other plugins:
* `kubernetes`
* `docker`
* `docker-workflow`
* `amazon-ecs`